lib/engine/company.rb

```diff
@@ -59,6 +59,10 @@ def corporation?
       false
     end
 
+    def find_token_by_type(token_type)
```

Comment:
should theis instead check for the company ability for token? you don't want every company creating tokens right?

Reply:
This was why I went with the first approach to be creating the token in 1 spot for the ability code. I'll add another check here and raise a game error if not.